### PR TITLE
Fix pinning sidebar filters for group datasets

### PR DESCRIPTION
**Pull request by benjaminpkane**
## What changes are proposed in this pull request?

Adding sidebar label filters to the current view in group datasets currently omits label filtering that one might expect. This was due to filtering occurring before group slices were selected, causing the filtering rules to be overwritten by the group lookup stage

https://github.com/voxel51/fiftyone/assets/19821840/3486169f-60de-4a93-8278-a9bf70ab19c3

## How is this patch tested? If it is not, please explain why.

In progress

## Release Notes

* Fixed label filtering after adding sidebar filters to the current view for group datasets

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Enhanced filtering capabilities in the data view interface.
- **Refactor**
	- Improved the logic for setting slice views based on conditions related to form parameters.
	- Refactored the handling of sample filter groups for better selection of group slices and IDs.
	- Updated the creation of saved views to exclude unnecessary sample filter conditions.
- **Tests**
	- Updated unit tests to accommodate changes in view functions and optimizations.

### CODE REVIEW

File: fiftyone/core/stages.py [hunk 1]
```diff
@@ -941,8 +942,8 @@ def _get_meta_filtered_fields(sample_collection, meta_filter, frames=False):
             base, leaf = key.split(".", 1)
             info_filter[leaf] = val
 
-    matcher = (
-        lambda q, v: q.lower() in v.lower()
+    matcher = lambda q, v: (
```
Comment (swheaton) [inline fiftyone/core/stages.py:945]: I find this lambda incredibly confusing. I recommend refactoring to a named function instead
```python

  def matcher(query, value):
    if isinstance(query, str):
      if isinstance(value, str):
        return query.lower() in value.lower()
      elif isinstance(value, dict):
        return query.lower() in str(value).lower()
    return query == value
```

File: fiftyone/core/stages.py [hunk 2]
```diff
@@ -951,8 +952,8 @@ def _get_meta_filtered_fields(sample_collection, meta_filter, frames=False):
         )
     )
 
-    type_matcher = (
-        lambda query, field: (
+    type_matcher = lambda query, field: (
```
Comment (swheaton) [inline fiftyone/core/stages.py:955]: Same, I find this lambda incredibly confusing. I recommend refactoring to a named function instead

```python
  def type_matcher(query, field):
    if isinstance(query, str):
      if isinstance(field, EmbeddedDocumentField):
        return type(field.document_type).__name__ == query or field.document_type.__name__ == query
      else:
        return type(field).__name__ == query
    elif isinstance(field, EmbeddedDocumentField):
      return isinstance(field.document_type, query) or field.document_type == query
    else:
      return isinstance(field, query)
```

File: fiftyone/server/aggregations.py [hunk 3]
```diff
@@ -132,7 +132,26 @@ async def aggregate_resolver(
...
+            extended_stages=form.extended_stages,
+            sample_filter=SampleFilter(
+                group=(
+                    GroupElementFilter(
+                        id=form.group_id, slice=form.slice, slices=[form.slice]
+                    )
+                    if not form.sample_ids
+                    else None
+                )
+            ),
```
Comment (swheaton) [inline fiftyone/server/aggregations.py:150]: did you intend for slice view to not be awaitable like above view is?

File: fiftyone/server/aggregations.py [hunk 4]
```diff
@@ -132,7 +132,26 @@ async def aggregate_resolver(
         awaitable=True,
     )
 
-    slice_view = view if form.mixed and "" in form.paths else None
+    slice_view = (
+        fosv.get_view(
```
Comment (swheaton) [inline fiftyone/server/aggregations.py:136]: this is almost exactly the same as the call above. 1) is that right? 2) can we re-use it somehow to remain DRY?

File: fiftyone/server/view.py [hunk 5]
```diff
@@ -121,18 +121,43 @@ def run(dataset):
...
         if sample_filter is not None:
             if sample_filter.group:
-                if sample_filter.group.slice:
-                    view.group_slice = sample_filter.group.slice
-
-                if sample_filter.group.id:
-                    view = fov.make_optimized_select_view(
-                        view, sample_filter.group.id, groups=True
-                    )
+                unselected = all(
```
Comment (swheaton) [inline fiftyone/server/view.py:124]: large code block here, would appreciate some comments as I don't know exactly what's happening - let alone future me.

**Review by swheaton — CHANGES_REQUESTED** [review]

File: fiftyone/core/stages.py (hunk 2 above)
Comment (benjaminpkane) [reply fiftyone/core/stages.py:955]: The diff here and the one above are only due to formatting. I can update, but it has no affect on the focus of the PR

Comment (swheaton) [reply fiftyone/core/stages.py:955]: I know. Your choice on whether you want to refactor in this PR or not. Just calling it out.

File: fiftyone/server/aggregations.py (hunk 3 above)
Comment (benjaminpkane) [reply fiftyone/server/aggregations.py:150]: It should be awaitable. Thanks 👍

**Review by swheaton — APPROVED** [review]
lgtm after changes

**Pull request merged**